src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -319,10 +319,22 @@ def update_container_settings(resource_group_name, name, docker_registry_server_
     settings = []
     if docker_registry_server_url is not None:
         settings.append('DOCKER_REGISTRY_SERVER_URL=' + docker_registry_server_url)
+
+    if (not docker_registry_server_user and not docker_registry_server_password and
+            docker_registry_server_url and '.azurecr.io' in docker_registry_server_url):
+        logger.warning('No credential was provided to access Azure Container Registry. Trying to look up...')
+        parsed = urlparse(docker_registry_server_url)
+        registry_name = (parsed.netloc if parsed.scheme else parsed.path).split('.')[0]
+        try:
+            docker_registry_server_user, docker_registry_server_password = _get_acr_cred(registry_name)
+        except Exception as ex:  # pylint: disable=broad-except
+            logger.warning("Error:'%s'", ex)  # consider to throwing if needed
```

Comment:
`logger.warning` and `"Error"` in the actual message is confusing.
Can you use `logger.error` instead? Or, if this is a fatal error, raise CLIError.